pennylane/transforms/core/transform_dispatcher.py

```diff
@@ -60,9 +60,8 @@ def __call__(self, *targs, **tkwargs):
             # is the object we wish to transform
             obj, *targs = targs
 
-        if isinstance(obj, qml.tape.QuantumTape):
-            new_tape = copy.deepcopy(obj)
-            return self._transform(new_tape, *targs, **tkwargs)
+        if isinstance(obj, qml.tape.QuantumScript):
+            return self._transform(obj, *targs, **tkwargs)
```

Comment:
why don't we need the copy anymore? handled in transforms automatically, or perhaps it doesn't mutate?

Reply:
I think it will be also impossible to mutate a tape in the future (pytree)